File: Components/DevExpressVCL/Demos/ExpressBars/Delphi/InPlaceEditorsDemo/InPlaceEditorsDemoImage.pas
unit InPlaceEditorsDemoImage;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, ExtCtrls, InPlaceEditorsDemoFrameManager, cxControls, cxContainer,
  cxEdit, cxImage, StdCtrls;

type
  TfrmImageEditors = class(TEditorDemoBaseFrame)
    cxImage1: TcxImage;
  private
    { Private declarations }
  public
    procedure SetParameters(AStream: TStream);
  end;

implementation

{$R *.dfm}

{ TfrmImageEditors }

procedure TfrmImageEditors.SetParameters(AStream: TStream);
begin
  if AStream <> nil then
    cxImage1.Picture.Bitmap.LoadFromStream(AStream)
  else
    cxImage1.Picture.Bitmap := nil;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressBars/CBuilder/InPlaceEditorsDemo/InPlaceEditorsDemoImage.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "InPlaceEditorsDemoImage.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxEdit"
#pragma link "cxImage"
#pragma resource "*.dfm"
TfrmImageEditors *frmImageEditors;
//---------------------------------------------------------------------------
__fastcall TfrmImageEditors::TfrmImageEditors(TComponent* Owner)
	: TEditorDemoBaseFrame(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmImageEditors::SetParameters(TStream* AStream)
{
  if (AStream != NULL)
    cxImage1->Picture->Bitmap->LoadFromStream(AStream);
  else
    cxImage1->Picture->Bitmap = 0;
}

